client/state/document/reducer.js

```diff
@@ -0,0 +1,51 @@
+/**
+ * External dependencies
+ */
+import { combineReducers } from 'redux';
+
+/**
+ * Internal dependencies
+ */
+import {
+	SET_DOCUMENT_DESCRIPTION,
+	SET_DOCUMENT_TITLE,
+} from 'state/action-types';
+
+const INITIAL_STATE = {
+	title: '',
+	description: '',
+};
+
+export function title( state = '', action ) {
+	switch ( action.type ) {
+		case SET_DOCUMENT_TITLE:
+			return action.title;
+	}
+
+	return state;
+}
+
+export function description( state = '', action ) {
+	switch ( action.type ) {
+		case SET_DOCUMENT_DESCRIPTION:
+			return action.description;
+	}
+
+	return state;
+}
+
+export function documentSettings( state = INITIAL_STATE, action ) {
+	switch ( action.type ) {
+		case SET_DOCUMENT_TITLE:
+			return Object.assign( {}, state, { title: title( state.title, action ) } );
+
+		case SET_DOCUMENT_DESCRIPTION:
+			return Object.assign( {}, state, { description: description( state.description, action ) } );
+	}
+
+	return state;
+}
+
+combineReducers( {
```

Comment:
This is not effectively doing anything. I assume you want to export it as a default. Also keep in mind that when you do, the `documentSettings` function state will be accessible from `state.documentSettings.documentSettings` (after wiring into `state/index.js`) which is obviously redundant.

I'd suggest getting rid of `documentSettings` function altogether, and changing this to export the `title` and `description` functions, each of which are only responsible for managing the state of their corresponding strings.